### PR TITLE
Story/cord 20

### PR DESCRIPTION
**Pull request by aa-kashk**

### CODE REVIEW

File: cord19-explorer/pom.xml [hunk 1]
```diff
@@ -14,13 +14,19 @@
...
-
 		<admin.password>$2a$04$oQo44vqcDIFRoYKiAXoNheurzkwX9dcNmowvTX/hsWuBMwijqn44i</admin.password>
 
 		<db.driver>com.mysql.jdbc.Driver</db.driver>
 		<db.database.url>jdbc-url-to-db</db.database.url>
 		<db.user>db-user</db.user>
 		<db.password>db-password</db.password>
+		
+		<email.user></email.user>
+        <email.password></email.password>
```
Comment (Sagar-k93) [inline cord19-explorer/pom.xml]: Indentation

File: cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/config/SpringWebConfig.java [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 import org.springframework.web.servlet.config.annotation.EnableWebMvc;
 import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
 @Configuration
 @EnableWebMvc
-@ComponentScan
 public class SpringWebConfig implements WebMvcConfigurer {
 
     @Override
```
Comment (prjadhav14) [inline cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/config/SpringWebConfig.java:12]: remove line 12

File: cord19-explorer/src/main/webapp/WEB-INF/views/home.html [hunk 3]
```diff
@@ -37,7 +37,6 @@ <h4>Journals (total [[${journalCount}]])</h4>
 	     </div>
    		</div>
```
Comment (prjadhav14) [inline cord19-explorer/src/main/webapp/WEB-INF/views/home.html]: indentation

File: cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 4]
```diff
@@ -148,4 +148,25 @@ <h1>Ups, something went wrong.</h1>
 	    
    </div>
   </body>
+  <footer class="footer">
+      <div class="container">
+        <div class="row">
+        <div class="col-md-12">
+		<hr style="margin-bottom: 25px;">
+		<p class="text-muted pull-left">
+	    <p class="text-muted">
```
Comment (prjadhav14) [inline cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html]: indentation

**Review by prjadhav14 — CHANGES_REQUESTED** [review]
inline comments

File: cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 5]
```diff
@@ -148,4 +148,25 @@ <h1>Ups, something went wrong.</h1>
...
+	   	<div sec:authorize="isAnonymous()">
+		<form name='f' class="form-inline pull-right" th:action="@{/login/authenticate}" method="POST">
```
Comment (Sagar-k93) [inline cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html]: Use a useful name for the form.

File: cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html (hunk 4 above)
Comment (Sagar-k93) [inline cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html]: Indentation

File: cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 6]
```diff
@@ -148,4 +148,25 @@ <h1>Ups, something went wrong.</h1>
 	    
    </div>
   </body>
+  <footer class="footer">
+      <div class="container">
+        <div class="row">
+        <div class="col-md-12">
+		<hr style="margin-bottom: 25px;">
+		<p class="text-muted pull-left">
```
Comment (Sagar-k93) [inline cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html]: This para tag is never closed?

**Comment by jdamerow** [comment]
> username, password: admin

this is not the password for the staging server, but if it were you should not post it on a public site...

**Review by prjadhav14 — APPROVED** [review]

File: cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/config/SecurityContext.java [hunk 7]
```diff
@@ -22,12 +22,15 @@ public void configure(WebSecurity web) throws Exception {
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.formLogin().and().logout().deleteCookies("JSESSIONID").logoutSuccessUrl("/").and().exceptionHandling()
-                .accessDeniedPage("/403")
+        http.formLogin().loginPage("/").loginProcessingUrl("/login/authenticate").failureUrl("/?error=bad_credentials")
```
Comment (jdamerow) [inline cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/config/SecurityContext.java]: let's leave the login page on a separate page as it was

**Review by jdamerow — CHANGES_REQUESTED** [review]
I'm fairly certain you need to provide a register page and user management page

File: cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html [hunk 8]
```diff
@@ -43,7 +43,7 @@
 	      	<nav>
 	          <ul class="nav nav-pills pull-right">
 	          	
-	          	<li role="presentation">
+	          	<li  sec:authorize="hasAnyRole('ADMIN','USER')" role="presentation">
```
Comment (jdamerow) [inline cord19-explorer/src/main/webapp/WEB-INF/views/layouts/main.html]: that should stay public as it was before

**Review by jdamerow — CHANGES_REQUESTED** [review]
There is no link to the user admin page where users can be approved, etc. I'm fairly certain it also needs a jsp apge.

**Comment by aa-kashk** [comment]
Isn't this the user admin page?
image

File: cord19-explorer/pom.xml (hunk 1 above)
Comment (jdamerow) [reply cord19-explorer/pom.xml]: this

**Comment by jdamerow** [comment]
yes, it is. never mind.

File: cord19-explorer/pom.xml (hunk 1 above)
Comment (jdamerow) [reply cord19-explorer/pom.xml]: ?

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
reset password link doesn't work (just goes back to login page)

**Comment by jdamerow** [comment]
also, emails are not being sent. is that working locally for you?

**Comment by aa-kashk** [comment]
The emails for the reset password link? No. I will look into it.

**Comment by jdamerow** [comment]
any emails.

File: cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestController.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class ResetPasswordRequestController {
+
+    @RequestMapping(value="/reset/password/request")
+    public String show(Model model) {
+        return "resetPassword";
+    }
+}
```
Comment (jdamerow) [inline cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestController.java:14]: I believe this controller is provided by the simple user plugin?
https://github.com/diging/simple-users/tree/develop/simple-users/src/main/java/edu/asu/diging/simpleusers/web

File: cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestSentController.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class ResetPasswordRequestSentController {
+
+    @RequestMapping("/reset/request/sent")
+    public String show() {
+        return "resetRequestSent";
+    }
+}
```
Comment (jdamerow) [inline cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestSentController.java:13]: same here?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestController.java (hunk 9 above)
Comment (aa-kashk) [reply cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestController.java:14]: I will remove these controllers and see if it works.

Comment (jdamerow) [reply cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/ResetPasswordRequestController.java:14]: you probably have to configure it with the urls

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
- when I try to create an account with a username that already exists, I only get redirected back to the signup page but no error message
- do emails get sent when you run it locally? the staging server does not send any emails.

**Comment by aa-kashk** [comment]
- Error messages are being displayed. The issue was with the thymeleaf template. Debugging showed errors were being updated correctly. 

- For the email issue, the issue is with the config in pom.xml. The pom.xml containing config info is empty which is why the emails are not being sent.

**Comment by jdamerow** [comment]
the email configuration doesn't work; my guess is that this is the case because you are missing the propertysource annotation on the config class, so the email configuration values are never set in the config class.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
It's still throwing an error. Have you tried it locally? Does it send an email?

**Comment by aa-kashk** [comment]
when I configure pom.xml with <email.host>gmail.com</email.host>, it sets up the javamailsender. Currently, my pom xml doesn't have these values, its an empty string(<email.host></email.host>), the mailSender is not being setup on startup due to this.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
please resolve conflicts

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Alright, finally figured it out. Please update simple-user dependencies to 1.1 and sending email finally work.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
clicking on the reset link in the email throws this exception:
```
java.io.FileNotFoundException: Could not open ServletContext resource [/WEB-INF/views/password/change.html]
	org.springframework.web.context.support.ServletContextResource.getInputStream(ServletContextResource.java:159)
	org.thymeleaf.spring5.templateresource.SpringResourceTemplateResource.reader(SpringResourceTemplateResource.java:103)
	org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parse(AbstractMarkupTemplateParser.java:223)
	org.thymeleaf.templateparser.markup.AbstractMarkupTemplateParser.parseStandalone(AbstractMarkupTemplateParser.java:100)
	org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:666)
	org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1098)
	org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1072)
	org.thymeleaf.spring5.view.ThymeleafView.renderFragment(ThymeleafView.java:362)
	org.thymeleaf.spring5.view.ThymeleafView.render(ThymeleafView.java:189)
	org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1373)
	org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1118)
	org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1057)
	org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
	org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
	org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
	org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	
```

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
all the password simple user pages have to be adjusted to the new layout
image
image
and the email sent and change password pages as well.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
I'll merge this for now, but please adjust the label and input field alignment (labels should not be glued to the input field above).
image

**Pull request merged**